From 60b0cf2fc2fdc74d13832d1fdde21cfdc813b84e Mon Sep 17 00:00:00 2001
From: Emmanuele Bassi <ebassi@gnome.org>
Date: Mon, 26 Apr 2021 10:59:52 +0100
Subject: [PATCH] scanner: Add more float types

The scanner is chocking on the `__float80` and `__float128` types that
are provided by GCC as extensions to ISO/IEC TS 18661-3:2015:

https://gcc.gnu.org/onlinedocs/gcc/Floating-Types.html

Fixes: #384
---
 giscanner/scannerlexer.l | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/giscanner/scannerlexer.l b/giscanner/scannerlexer.l
index cfec4d58..b6cc6499 100644
--- a/giscanner/scannerlexer.l
+++ b/giscanner/scannerlexer.l
@@ -224,6 +224,8 @@ stringtext				([^\\\"])|(\\.)
 "_Float32x"                             { return BASIC_TYPE; }
 "_Float64x"                             { return BASIC_TYPE; }
 "_Float128x"                            { return BASIC_TYPE; }
+"__float80"                             { return BASIC_TYPE; }
+"__float128"                            { return BASIC_TYPE; }
 "for"					{ return FOR; }
 "goto"					{ return GOTO; }
 "if"					{ return IF; }
-- 
2.31.1

